### PR TITLE
Updating teleport-quickstart.yml to latest release

### PR DESCRIPTION
**Pull request by corkrean**
This teleport-quickstart.yml is used in our Docker quickstart guide (docker-compose section). https://goteleport.com/docs/quickstart-docker/#quickstart-using-docker-compose

I updated the quay repo from 5.0 to 6.1.5. (latest version at time of writing)

### CODE REVIEW

File: docker/teleport-quickstart.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ services:
   # The configure container starts, generates a config, writes it to
   # /etc/teleport/teleport.yaml and then immediately exits.
   configure:
-    image: quay.io/gravitational/teleport:5
+    image: quay.io/gravitational/teleport:6.1.5
```
Comment (webvictim) [inline docker/teleport-quickstart.yml:6]: The old `5` tag was a 'nightly' tag - i.e it is rebuilt every night and points to the latest `5.x` Teleport version. We should use the equivalent `6` tag here.

[Here's a list of tags](https://quay.io/repository/gravitational/teleport?tag=latest&tab=tags) - see the `4.4`, `5` and `6` at the top.

```suggestion
    image: quay.io/gravitational/teleport:6
```

File: docker/teleport-quickstart.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ services:
   # This container depends on the config written by the configure container above, so it
   # sleeps for a second on startup to allow the configure container to run first.
   teleport:
-    image: quay.io/gravitational/teleport:5
+    image: quay.io/gravitational/teleport:6.1.5
```
Comment (webvictim) [inline docker/teleport-quickstart.yml:17]:
```suggestion
    image: quay.io/gravitational/teleport:6
```

**Review by webvictim — CHANGES_REQUESTED** [review]

**Comment by webvictim** [comment]
@corkrean As you raised this PR directly against the Teleport repo, the tests will run and you'll be able to merge it yourself. See the "checks" section above and the link to Drone CI: https://drone.gravitational.io/gravitational/teleport/11891

**Review by corkrean — COMMENTED** [review]
Changing to incorporate nightly change repo.

**Review by webvictim — APPROVED** [review]

**Review by inertial-frame — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Comment by webvictim** [comment]
Needs to be backported along with #7055

**Pull request merged**